env/profiling/hitchhiker/bench.clj

```diff
@@ -194,74 +196,77 @@
-            tree-to-test (atom {})
-            results (atom [])]
-        (cond
-          (or (= "-h" root)
-              (= "--help" root)
-              (nil? root)
-              (:help options)) (exit 0 (usage summary))
-          (not= (count arguments) 0) (exit 1 (usage summary))
-          errors (exit 1 (error-msg errors)))
-        (let [backend (case (:backend options)
-                        "testing" (core/->TestingBackend)
-                        "redis" (do (redis/start-expiry-thread!)
-                                    (redis/->RedisBackend)))
-              delete-xform (case (:delete-pattern options)
-                             "forward" identity
-                             "reverse" reverse
-                             "shuffle" shuffle
-                             "zero" #(repeat (count %) 0.0))
-              [tree-name structure]
-              (case (:data-structure options)
-                "b-tree" ["b-tree" (core-b-tree (:tree-width options) backend)]
-                "fractal" ["fractal" (msg-b-tree (:tree-width options) backend)]
-                "sorted-set" ["sorted-set" (sorted-set-repr)])
-              flush-freq (:flush-freq options)
-              codename (str tree-name
-                            "__flush_"
-                            flush-freq
-                            "__b_"
-                            (:tree-width options)
-                            "__"
-                            (:backend options)
-                            "__n_"
-                            (:num-operations options)
-                            "__del_"
-                            (:delete-pattern options))]
-          (doseq [ds (generate-test-datasets)
-                  :let [codename (str codename
-                                      "_"
-                                      (:name ds))
-                        out (create-output-dir
-                              root
-                              codename)
-                        _ (println "Doing" codename)
-                        bench-res (benchmark (:num-operations options) ds flush-freq structure out delete-xform)]]
-            (swap! results conj
-                   {:tree tree-name
-                    :ds (:name ds)
-                    :freq flush-freq
-                    :n (:num-operations options)
-                    :b (:tree-width options)
-                    :delete-pattern (:delete-pattern options)
-                    :results bench-res}))
-          ;(println "results")
-          ;(clojure.pprint/pprint @results)
-          (swap! outputs conj (template-one-sheet @results)))))
-    (excel/render-to-file
-      "template_benchmark.xlsx"
-      (.getPath (File. root "analysis.xlsx"))
-      {"SingleDS"
-       (map-indexed (fn [i s]
-                      (assoc s :sheet-name (str "Trial " (inc i))))
-                    @outputs)})))
+  (jdbc/with-db-connection [db (sqlite/db-spec ":memory:")]
```

Comment:
Could we move this down to the creation of the `SQLiteBackend`? I think that would make it easier to review by not changing the indent, and that's the only time we need the DB to be created.

Also, what are your thoughts on using memory vs. disk-backed DB. I feel like backing with disk is more interesting to trigger the IO behavior we might encounter.